### PR TITLE
Fix Issues #33 and #34

### PR DESCRIPTION
**Pull request by irhel**

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ Apify.main(async () => {
     // Prepare the initial list of google shopping queries and request queue
     const requestList = await makeRequestList(queries, inputUrl, countryCode);
     log.info('Search URLs:');
-    requestList.requests.forEach((r) => console.log('  ', r.url));
+    requestList.requests.forEach(r => console.log('  ', r.url));
```
Comment (metalwarrior665) [inline src/main.js]: This change is wrong, in our ES lint config, there should be parens

File: src/main.js [hunk 2]
```diff
@@ -45,11 +45,8 @@ Apify.main(async () => {
         proxyConfiguration,
         launchContext: {
             launchOptions: {
-                waitUntil: 'load',
-                // headless: true,
+                headless: true,
```
Comment (metalwarrior665) [inline src/main.js]: Don't force headless mode. It is nice to have it off locally. By default it is on on platform and off locally

File: src/routes.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+const { utils } = require('apify');
```
Comment (metalwarrior665) [inline src/routes.js]: You don't need to require apify twice

File: src/routes.js [hunk 4]
```diff
@@ -29,10 +26,7 @@ exports.SEARCH_PAGE = async (page, request, query, requestQueue, maxPostCount, e
             '#debug': Apify.utils.createRequestDebugInfo(request),
         });
     }
-    // sometimes it loads totally different page. In this case number of items would be > 22. Need to retry.
-    if (resultsLength > 22) {
-        throw new Error("Page didn't load properly, retrying...");
-    }
```
Comment (metalwarrior665) [inline src/routes.js:35]: Are you sure this is ok to remove?

File: src/utils.js [hunk 5]
```diff
@@ -20,7 +20,7 @@ function checkAndEval(extendOutputFunction) {
 }
 
 async function applyFunction(page, extendOutputFunction, item) {
-    const isObject = (val) => typeof val === 'object' && val !== null && !Array.isArray(val);
+    const isObject = val => typeof val === 'object' && val !== null && !Array.isArray(val);
```
Comment (metalwarrior665) [inline src/utils.js]: Again, no change pls

**Review by metalwarrior665 — CHANGES_REQUESTED** [review]
What was actual the issue? The only significant change I see is the < 22 items.

**Comment by irhel** [comment]
> What was actual the issue? The only significant change I see is the < 22 items.

Yup. Throws an error when the number of results is greater than 22 which is the case for the first page of results.

File: src/routes.js (hunk 4 above)
Comment (irhel) [reply src/routes.js:35]: Yes, it throws an error and the exact error is reported in #33 and #34 .

**Pull request closed without merging**